fecgov/fec-cms
# Launch Sitemaps for site-search

## Summary (required)
Launch new global sitemap configuration:
- We are transitioning away from search.gov's i14Y drawer to using sitemaps indexed by search.gov.

Changes to policy-guidance sitemaps:
- We make changes to policy-guidance search configuration, which currently uses sitemaps, in order for both to now use sitemaps indexed by seacrh.gov

See [issue summary](https://github.com/fecgov/fec-cms/issues/5775#issue-1765493096) for detailed explanation of sitemaps and robots.txt

- Resolves issue  #5775 


- [x] Check with search.gov as to whether there are any changes they need to make on their end when we deploy:
    - Does search.gov have to adjust anything when we delete the i14Y drawer to start  using sitemaps referenced in `robots.txt` instead?
    **Answer:**  Once we deploy, they will kickoff a manual  indexing of the new `/sitemap-wagtail.xml`  and manually change affiliate to  use that index instead of i14y drawer.    
    - Does search.gov have to adjust anything because now the policy-guidance sitemaps are referenced by `robots.txt`. When we originally setup policy-guidance , we had to tell them them to start indexing these sitemaps regularly because there was no reference to them in `robots.txt`(or anywhere else).
    **Answer:** Search.gov will do a manual index of sitemaps upon deploy. Thereafter, sitemaps will be indexed on schedule.   They will delete internal setting no longer needed  to reference these  sitemaps

**The steps below need to be coordinated to happen at same time as the deployment of the release.  Also requires  coordination with search.gov team**
 **Once deploy is live, follow these steps:** 
- [ ]  If necessary, Change site settings in Wagtail admin ( Settings > Sites ) to `host: fec.gov` , `port: 443`
- [ ] Change domains in search.gov [fec_content_s3_prod](https://search.usa.gov/sites/8042)  to `www.fec.gov/resources/cms-content/documents/policy-guidance`  --and--   `www.fec.gov/updates/guidance-search`
- [ ] In Wagtail documents, replace `sitemap_pdf.xml` and `sitemap_html.xml` with the new ones in this PR.
- [ ] Alert sesrch.gov to kickoff manual index of all three sitemaps listed in `robots.txt`.
- [ ]  Add these to `Best-Bets` in [betafec_api](https://search.usa.gov/sites/6738), if not already there: Are there any missing from this list ?
    - https://www.fec.gov/calendar/
    - https://www.fec.gov/about/leadership-and-structure/commissioners/
    - https://www.fec.gov/help-candidates-and-committees/question-rad/
    - https://www.fec.gov/help-candidates-and-committees/guides/
    - https://www.fec.gov/meetings/
    - https://www.fec.gov/legal-resources/policy-and-other-guidance/guidance-documents/
    - https://www.fec.gov/updates/
 - [ ]  In `Best-Bets` in the  [fec_content_s3_prod](https://search.usa.gov/sites/146954/) affiliate,  change any policy PDF's to have the new `/policy-guidance/` directory in their paths. Change any webpages to have `/guidance-search` in their path. 
- [ ]  After making edits, double-check  that  the keys from the Activate (</>) tab on both search.gov affiliates have not changed and are the same as whats  referenced in `base.py` and `search/views.py`.
- [ ] Turn  password protection back on for dev.fec.gov home page (which password protects all its chid wagtail pages)
- [ ]  After confirming global site-search works as expected, delete i14y drawer from [betafec_api](https://search.usa.gov/sites/6738)
- [ ] Remove `SEARCHGOV_DRAWER_HANDLE` and 'SEARCHGOV_DRAWER_KEY_MAIN'  env vars from CloudFoundry, User Provided Service.
- [ ]  Remind search.gov  to delete their internal settings that point to the policy-guidance sitemaps because they are now referenced by `robots.txt`


### Required reviewers

one or two developers (frontend/backend)  one content

## Impacted areas of the application

This effects site search and policy-guidance search handled by search.gov. It is not related to external search engines SEO.

	modified:   fec/fec/settings/base.py
	modified:   fec/fec/templates/robots_prod.txt
	modified:   fec/fec/tests/test_robots.py
	modified:   fec/fec/urls.py
	modified:   fec/home/wagtail_hooks.py
	modified:   fec/search/management/data/sitemap_html.xml
	modified:   fec/search/management/data/sitemap_pdf.xml
	modified:   fec/search/tests/test_views.py
	modified:   fec/search/views.py

## Screenshots

**One notable improvement is that results show the SEO title instead of  just page title**
<hr>

### BEFORE:
<img width="791" alt="Screen Shot 2023-08-10 at 11 08 02 AM" src="https://github.com/fecgov/fec-cms/assets/5572856/5f7aec8d-1ca3-46b4-9a6b-985f4d30ed2b">

<hr>

### AFTER:

<img width="742" alt="Screen Shot 2023-08-10 at 11 07 03 AM" src="https://github.com/fecgov/fec-cms/assets/5572856/51d5513c-bc70-4006-a50e-30f9e074296a">


<hr>
## Related PRs
https://github.com/fecgov/fec-cms/pull/5735

## How to test
- checkout branch
- run `pytest` from `fec-cms` directory and from `fec` directory 
-  `./manage.py runserver`  
- see Wagtail sitemap at `http://127.0.0.1:8000/sitemap-wagtail.xml/`
- `export FEC_CMS_ENVIRONMENT=prod` to see prod robots.txt at `http://127.0.0.1:8000/robots.txt`
- Don't forget to set your env back to local:  `export FEC_CMS_ENVIRONMENT=local`

Site-search results and sitemap indexing were tested extensively on dev space pointing to  test affiliates in search.gov.  Check with @johnnyporkchops to see if the test branch is deployed to dev if you want to test live site-search results. For background  Pease see  [PR](https://github.com/fecgov/fec-cms/pull/5735).  

If the test branch is still deployed to dev, you can  also test search results in the search.gov preview at : https://search.usa.gov/search?affiliate=fec_content_s3_dev
https://search.usa.gov/search?affiliate=dev-fec.gov
